components/rendering/src/lib.rs

```diff
@@ -2,21 +2,28 @@ mod context;
 mod markdown;
 mod shortcode;
 mod table_of_contents;
+mod extract_katex;
 
 use errors::Result;
 
 pub use context::RenderContext;
+pub use extract_katex::render_katex;
 use markdown::markdown_to_html;
 pub use shortcode::render_shortcodes;
 pub use table_of_contents::Heading;
 
+
 pub fn render_content(content: &str, context: &RenderContext) -> Result<markdown::Rendered> {
+    let mut preprocessed: String = content.to_string();
+
     // Don't do shortcodes if there is nothing like a shortcode in the content
-    if content.contains("{{") || content.contains("{%") {
-        let rendered = render_shortcodes(content, context)?;
-        let html = markdown_to_html(&rendered, context)?;
-        return Ok(html);
+    if preprocessed.contains("{{") || preprocessed.contains("{%") {
+        preprocessed = render_shortcodes(&preprocessed, context)?;
     }
 
-    markdown_to_html(&content, context)
+    preprocessed = render_katex(&preprocessed);
```

Comment:
this needs to be gated on some config option, I can add it myself later. Any preference on the name? `katex = true`?

Reply:
yeah sure, or `render_latex_math = true`?

Reply:
Since katex and mathjax have slightly different feature sets, maybe make it clear that rendering rely on katex?
`katex = true` seems a good choice.

Reply:
I'd also prefer `katex = true` (Does MathJax require processing at render time?)